src/ex4/dist/style.css

```diff
@@ -0,0 +1,264 @@
+    width: 60%;
+    max-height: 80%;
+    margin: 50px auto;
+    padding: 20px;
+    border-radius: 5px;
+    box-shadow: 0px 10px 15px rgba(0,0,0,0.1);
+}
+
+.top {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+}
+
+.filter-select {
+    width: 200px;
+    padding: 10px;
+    border-radius: 5px;
+    border: none;
+    font-size: 16px;
+    font-weight: 500;
+    background-color: rgba(182, 128, 191, 0.3);
+    text-align: center;
+}
+
+.filter-select:focus{
+    outline: none;
+}
+
+.title {
+    font-size: 30px;
+    font-weight: bold;
+}
+
+.input-container {
+    display: flex;
+    margin: 20px 0;
+    height: 45px;
+    width: 100%;
+    justify-content: space-between;
+}
+
+.todo-input {
+    width: 85%;
+    height: 100%;
+    outline: none;
+    border-radius: 5px;
+    border: 1px solid #ccc;
+    padding-left: 15px;
+    font-size: 16px;
+    transition: all 0.3s ease;
+}
+
+.todo-input:focus {
+    border: 2px solid #5f56c4;
+}
+
+.button {
+    background: #8E49E8;
+    color: #fff;
+    border-radius: 5px;
+    cursor: not-allowed;
+    border: none;
+    outline: none;
+    font-size: 20px;
+    font-weight: bold;
+    opacity: 0.6;
+    transition: all 0.3s ease;
+}
+
+.add-button {
+    width: 10%;
+    min-width: 35px;
+    height: 50px;
+    padding: 0;
+    margin-left: 5px;
+}
+
+.add-button.active, .clear-all-button.active {
+    opacity: 1;
+    transform: scale(0.98);
+    cursor: pointer;
+}
+
+.empty-todos-message {
+    background-image: url("https://www.itgovernance.eu/blog/de/wp-content/uploads/2017/04/list.jpg");
+    background-size: cover;
+    height: 40vh;
+}
+
+.enter-todo-title {
+    text-align: center;
+    vertical-align: middle;
+    line-height: 40vh;
+    z-index: 100;
+}
+
+.todo-list-container {
+    max-height: 250px;
+    overflow-y: auto;
+    padding: 0;
+}
+
+.action-btn{
+    width: 50px;
+    text-align: center;
+    color: #fff;
+    padding: 10px;
+    cursor: pointer;
+    opacity: 0.2;
+    transition: all 0.5s ease;
+}
+
+.todo-item {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    list-style: none;
+    margin-bottom: 8px;
+    height: 25px;
+    background: #f2f2f2;
+    border-radius: 5px;
+    padding: 10px 0 10px 10px;
+    overflow: hidden;
+    word-wrap: break-word;
+    cursor: default;
+}
+
+.image {
+   height: 50px;
+}
+
+.actions{
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.delete{
+    background: #e74c3c;  
+}
+
+.edit{
+    background: #5e3ce7;  
+    border-radius: 0 5px 5px 0;
+}
+
+.todo-item:hover {
+    background: rgba(139, 90, 139, 0.3);
+}
+
+.todo-item:hover .action-btn{
+    opacity: 1;
+}
+
+.clear-and-pending-container {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    width: 100%;
+    margin-top: 20px;
+}
+
+.clear-all-button {
+    padding: 10px;
+    word-wrap: break-word;
+}
+
+.switch {
```

Comment:
there is too much going on here. try to separate into multiple files (@import) by responsibility (ex: app.style.css, todos.style.css, todo.style.css)